ramlfications/utils/types.py

```diff
@@ -63,4 +66,64 @@ def parse_type(name, raw, root):
             data.pop("properties")
         except KeyError:
             pass
+
+    data.update(parse_examples(root.raml_version, data))
+
     return data_type_cls(**data)
+
+
+def parse_examples(raml_version, data):
+    #
+    # This is the only place that knows when to supply examples and when
+    # not, and that checks when both are supplied.
+    #
+    # This is also responsible for generating structured Example objects
+    # when appropriate (RAML >= 1.0).
+    #
+    data = data or {}
+    kw = {}
+    example = data.get("example", None)
```

Comment:
The `None` is not needed - already defaults to `None` if it can't find it/is not set.